src/components/Modal/Modal.js

```diff
@@ -0,0 +1,183 @@
+    purchasedAmount: '',
+    date: ''
+  };
+
+  toggle = () =>
+    this.setState((prevState) => ({
+      isOpen: !prevState.isOpen
+    }));
+
+  clear = () => {
+    this.setState({
+      isOpen: false,
+      coin: '',
+      purchasedAmount: 0,
+      date: ''
+    });
+  };
+
+  handleClickOutside = (event) => {
+    if (this.state.isOpen && this.ref.current === event.target) {
+      this.toggle();
+      this.clear();
+    }
+  };
+
+  componentDidMount() {
+    document.addEventListener('mousedown', this.handleClickOutside);
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('mousedown', this.handleClickOutside);
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    console.log(this.state);
+  };
+
+  render() {
+    const currentCurrency = localStorage.selection;
+
+    const coinsDetails = this.props.coins.map((coin) => {
+      return { name: coin.name, id: coin.id };
+    });
+
+    const coinInformation = this.props.coins.filter(
+      ({ id }) => id === this.state.coin
+    );
+
+    const minimizedImage = coinInformation[0]?.image.replace('large', 'small');
+
+    return (
+      <>
+        <ModalButton onClick={this.toggle}>Add Asset</ModalButton>
+        {this.state.isOpen && (
+          <ModalOverlay ref={this.ref}>
+            <ModalContainer>
+              <ModalContent>
+                <ModalHeader>
+                  <ModalTitle>Select Coins</ModalTitle>
+                </ModalHeader>
+                <form onSubmit={this.handleSubmit}>
+                  <ModalBody>
+                    <LeftContent>
+                      <CoinImageContainer>
+                        <CoinImage
+                          src={minimizedImage}
+                          alt={coinInformation[0]?.id}
+                        />
+                      </CoinImageContainer>
+                      <CoinNameText>
+                        {coinInformation[0]?.name}
+                        {`(${coinInformation[0]?.symbol.toUpperCase()})`}
+                      </CoinNameText>
+                    </LeftContent>
+                    <RightContent>
+                      <StyledDropdown
+                        options={coinsDetails}
+                        isOptionEqualToValue={(props) => props.name}
+                        getOptionLabel={(option) => option.name}
+                        PaperComponent={StyledPaper}
+                        ListboxComponent={StyledListbox}
+                        renderInput={(params) => (
+                          <TextField
+                            {...params}
+                            name="coin"
+                            label="Select Coins"
+                          />
+                        )}
+                        onChange={(_, value) => {
+                          this.setState({
+                            ...this.state,
+                            coin: value?.id || ''
+                          });
+                        }}
```

Comment:
please create a class method, don't have this setstate method run in render.

this arrow function gets created everytime render is called, is a waste of resources